code/modules/reagents/reagent_containers/spray.dm

```diff
@@ -108,6 +107,18 @@
 	desc = "BLAM!-brand non-foaming space cleaner!"
 	list_reagents = list("cleaner" = 250)
 
+/obj/item/reagent_containers/spray/cleaner/safety
+	desc = "BLAM!-brand non-foaming space cleaner! This spray bottle can only accept space cleaner."
+
+/obj/item/reagent_containers/spray/cleaner/safety/on_reagent_change()
+	for(var/datum/reagent/R in reagents.reagent_list)
```

Comment:
Can be made typeless

Reply:
I've forgotten how to do typeless, it's been awhile.

Reply:
```dm
for(var/thing in list)
  var/datum/path/to/obj/O = thing
  O.do_stuff()
```
Will make sure that the list is not filtered. Since the reagent_list should always have reagents in it this should be fine